app/layout/scenarios/edit/solutions/overview/component.tsx

```diff
@@ -78,7 +85,30 @@ export const ScenariosSolutionsOverview: React.FC<ScenariosSolutionsOverviewProp
     (bestSolutionisFetching && !bestSolutionisFetched) ||
     (selectedSolutionisFetching && !selectedSolutionisFetched);
 
-  // const frequencyLegendValues = LEGEND_LAYERS.frequency().items;
+  const onDownloadSolutionsSummary = useCallback(() => {
+    setSolutionsReportLoader(true);
+    downloadSolutionsSummary.mutate(
+      { id: pid },
+      {
+        onSuccess: () => {
+          setSolutionsReportLoader(false);
+        },
+        onError: () => {
+          setSolutionsReportLoader(false);
```

Comment:
It's better moving `setSolutionsReportLoader(false);` to `onSettle` callback, it will trigger either the mutation success or fails.